src/main/java/by/exadel/internship/repository/InternshipRepository.java

```diff
@@ -2,10 +2,23 @@
 
 import by.exadel.internship.entity.Internship;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+import java.util.Optional;
 import java.util.UUID;
 
 @Repository
 public interface InternshipRepository extends JpaRepository<Internship, UUID> {
+
+    List<Internship> findAllByDeletedTrue();
+    List<Internship> findAllByDeletedFalse();
+    Optional<Internship> findAllByIdAndDeletedFalse(UUID internshipId);
+    Optional<Internship> findByIdAndDeletedTrue(UUID internshipId);
+
+    @Modifying
+    @Query("update Internship i set i.deleted = false WHERE i.id = :internshipId")
+    void activateInternshipById(UUID internshipId);
```

Comment:
** A query with named parameters is easier to read and is less error-prone in case the query needs to be refactored.
https://www.baeldung.com/spring-data-jpa-query

```suggestion
    void activateInternshipById(@Param( "internshipId" ) UUID internshipId);
```

Reply:
Fixed